### PR TITLE
MacOS signing

### PR DESCRIPTION
**Pull request by roman-bushuiev**

### CODE REVIEW

**Comment by roman-bushuiev** [comment]
Now macOS signing and notarization should work. Important side-changes:
- JavaFX is upgraded to 19.
- I commented out the distribution of `.dmg`. Currently only portable `.app` is distributed.
- `nz.ac.waikato.cms.weka:weka-stable:3.8.5` upgraded to `nz.ac.waikato.cms.weka:weka-dev:3.9.6`.
- The following "unused dependencies of dependencies" are removed: `netlib`, `jblas`, `jocl`, `sqlite`.

_Test Build fails because GitHub secrets are not visible in PRs._

**Review by SteffenHeu — APPROVED** [review]

**Review by SteffenHeu — COMMENTED** [review]
Do we need FX 19 or can it stay at 16?

**Comment by tomas-pluskal** [comment]
I think javafx 16 is not available for Apple M1, is it? We need to fix those JavaFX bugs anyway, I'll try to look into it over the holidays.

**Comment by roman-bushuiev** [comment]
> I think javafx 16 is not available for Apple M1, is it? We need to fix those JavaFX bugs anyway, I'll try to look into it over the holidays.

It is impossible to build MZmine on M1 with JavaFX < 19. However, once built via GitHub actions with JavaFX 16, MZmine works fine on M1. Probably we can keep JavaFX 16 for now.

**Comment by tomas-pluskal** [comment]
> 
> It is impossible to build MZmine on M1 with JavaFX < 19. However, once built via GitHub actions with JavaFX 16, MZmine works fine on M1. Probably we can keep JavaFX 16 for now.

I suppose the GitHub actions produce a build for the x64 platform, which then runs on M1 via Apple's x64 emulation layer. That's not ideal, even if it works for now.

File: build.gradle [hunk 1]
```diff
@@ -164,7 +188,7 @@ dependencies {
     implementation group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.5'
 //    implementation "io.github.msdk:msdk-id-sirius:$msdkVersion"
 
-    implementation "org.xerial:sqlite-jdbc:3.8.11.2"
+//    implementation "org.xerial:sqlite-jdbc:3.40.0.0"  // Is dependency to du-lab but is not needed for compilation
```
[the remark this answers is not in the document]
Comment (roman-bushuiev) [reply build.gradle]: I added a new commit keeping `implementation "org.xerial:sqlite-jdbc:3.40.0.0"`.

**Comment by roman-bushuiev** [comment]
Should we try to merge it and see if the newly built version works fine on macOS? 😺

**Comment by robinschmid** [comment]
Can you merge the latest master?
Did you try the action on you private repo?
I could also try on mine - but I don't have a mac at my disposal

File: build.gradle [hunk 2]
```diff
@@ -164,8 +188,8 @@ dependencies {
     implementation group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.5'
 //    implementation "io.github.msdk:msdk-id-sirius:$msdkVersion"
 
-    implementation "org.xerial:sqlite-jdbc:3.8.11.2"
     implementation "org.du-lab.adap:adap:4.1.10"
+    implementation "org.xerial:sqlite-jdbc:3.40.0.0"
```
Comment (robinschmid) [inline build.gradle:192]: @SteffenHeu can you check if that works for the tdf import?

**Comment by robinschmid** [comment]
I pushed this branch with some changes to they mzmine repo to run the workflow and it works... just need to check if the installer etc work

**Comment by tomas-pluskal** [comment]
I can test the installer if you point me to the location where I can find it

**Pull request closed without merging**